scripts/verify_adv_IM.py

```diff
@@ -32,79 +66,163 @@ def parse_args():
+                # if start and end are there, but no data = crashed
+                station_component_status = run_status.crashed.value
+            elif time_list.count(None) == 1:
+                # only starttime exist = wct timed out
+                station_component_status = run_status.not_finished.value
+                # else not started
+            else:
+                station_component_status = run_status.not_started.value
+            comp_mask = (df_model["station"] == station_name) & (
+                df_model["component"] == comp
+            )
+            df_model.loc[comp_mask, "status"] = station_component_status
+            df_model.loc[comp_mask, time_type.start_time.name] = time_list[
+                time_type.start_time.value
+            ]
+            df_model.loc[comp_mask, time_type.end_time.name] = time_list[
+                time_type.end_time.value
+            ]
+    #    print(f"{df_model}")
+    return df_model
 
-def main(im_calc_dir, adv_im_model, components):
 
-    failed = []
+def main(im_calc_dir, adv_im_model, components, simple_check=False, station_file=None):
 
+    #    df_station_status = pd.DataFrame(columns = column_names)
+    df_dict = []
     for model in adv_im_model:
         csv_path = os.path.join(im_calc_dir, "{}.csv".format(model))
-        # using try/except to prevent crash
-        try:
-            df = pd.read_csv(csv_path)
-        except FileNotFoundError:
-            failed.append((model, "NoCSV", None))
-            print("csv for {} does not exist".format(model))
-            continue
-        # check for null/nan
-        if df.isnull().values.any():
-            failed.append(
-                (model, "NullValue", df[df.isnull().values].station.to_list())
+
+        if station_file is not None:
+            station_list = load_station_file(station_file).index.tolist()
+        else:
+            # glob for station folders
+            # station_list = [ y for y in [os.path.join(im_calc_dir, x) for x in os.listdir(im_calc_dir)] if os.path.isdir(y) ]
+            station_list = [
+                x
+                for x in os.listdir(im_calc_dir)
+                if os.path.isdir(os.path.join(im_calc_dir, x))
+            ]
+        list_folders = [os.path.join(im_calc_dir, x) for x in station_list]
+
+        # initialize df with empty value with not started
+        df_model = pd.DataFrame(columns=column_names)
+        for component in components:
+            df_model = pd.concat(
+                [
+                    df_model,
+                    pd.DataFrame(
+                        {
+                            "station": station_list,
+                            "model": model,
+                            "component": component,
+                            "status": run_status.not_started.value,
+                        },
+                        columns=column_names,
+                    ),
+                ],
+                ignore_index=True,
```

Comment:
This can be refactored to use a list comprehension and one concat call. Off the top of my head concat is rather inefficient, so doing it once would be better